lib/libc/picolibc/Kconfig

```diff
@@ -7,7 +7,7 @@ config PICOLIBC_USE_MODULE
 	bool "Picolibc as module" if "$(TOOLCHAIN_HAS_PICOLIBC)" = "y"
 	default y if "$(TOOLCHAIN_HAS_PICOLIBC)" != "y"
 	depends on ZEPHYR_PICOLIBC_MODULE
-	depends on !GLIBCXX_LIBCPP
+	depends on !REQUIRES_FULL_LIBCPP
```

Comment:
This piece doesn't look like it's helping here -- you're making the use of the picolibc module *more* restrictive so that apps using any full standard c++ library can't use it. I think your fix would work without this change.

Reply:
reverted change.